packages/hardhat-core/test/internal/hardhat-network/provider/modules/hardhat.ts

```diff
@@ -2143,27 +2172,36 @@ describe("Hardhat module", function () {
             this.provider,
             "hardhat_setStorageAt",
             [DEFAULT_ACCOUNTS_ADDRESSES[0], "xyz", numberToRpcQuantity(99)],
-            'Errors encountered in param 1: Invalid value "xyz" supplied to : QUANTITY'
+            // TODO: https://github.com/NomicFoundation/edr/issues/104
+            `${
+              isEdr ? "" : "Errors encountered in param 1: "
+            }Invalid value "xyz" supplied to : QUANTITY`
           );
         });
 
         it("should reject a storage key that is greater than 32 bytes", async function () {
           const MAX_WORD_VALUE = 2n ** 256n;
-          await assertInvalidInputError(
+          await assertProviderError(
             this.provider,
             "hardhat_setStorageAt",
             [
               DEFAULT_ACCOUNTS_ADDRESSES[0],
               numberToRpcQuantity(MAX_WORD_VALUE + 1n),
               "0xff",
             ],
-            "Storage key must not be greater than or equal to 2^256. Received 115792089237316195423570985008687907853269984665640564039457584007913129639937."
+            // TODO: https://github.com/NomicFoundation/edr/issues/104
+            `Storage key must not be greater than or equal to 2^256. Received ${
+              isEdr
+                ? "0x10000000000000000000000000000000000000000000000000000000000000001"
+                : "115792089237316195423570985008687907853269984665640564039457584007913129639937"
+            }.`,
+            isEdr ? InvalidArgumentsError.CODE : InvalidInputError.CODE
```

Comment:
The error codes should match imo, as they are part of the public API.

Reply:
Found the answer on Slack:

> When returning deserialization errors, sometimes Hardhat returns an InvalidArgumentsError.CODE and other times it returns a InvalidInputError.CODE. In EDR, all errors returned from serde are homomorphous, preventing us from making a separation. Is there are reason why there are two error codes, 

I managed to make this work in my WIP PR for hardforks by reorganizing errors. Hoping to submit today. Imo it'd be important to make this work, as the error codes are part of the public API.

Reply:
Agreed that this is important, even more so than error messages, because code are probably checked programmatically, while checking an error message is known to be brittle in every platform.

Reply:
@alcuadrado or @fvictorio could you elaborate a little about the distinction between `InvalidInputError` and `InvalidArgumentError` and why sometimes an error in deserialization is considered the former and sometimes the latter?

Reply:
Resolved by https://github.com/NomicFoundation/hardhat/pull/4721/commits/c0db3d43a42224038f55ce71c35079fae80062ae